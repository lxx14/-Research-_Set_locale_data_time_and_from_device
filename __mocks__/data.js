import moment from "moment";

export const users = [
    {
        name: 'Vasya',
        online: 'online',
        id: 1,
        messages: [
            {
                id: 11,
                date: moment("20190820", "YYYYMMDD").fromNow(),
                text: 'hello',
                recieved: false
            },
            {
                id: 12,
                date: moment("20190822", "YYYYMMDD").fromNow(),
                text: 'hi bro',
                recieved: true
            }
        ]
    },
    {
        name: 'Petya',
        online: 'last seen just now',
        id: 2,
        messages: [
            {
                id: 21,
                date: moment("20190821", "YYYYMMDD").fromNow(),
                text: 'Have a nice day!',
                recieved: false
            },
            {
                id: 22,
                date: moment("20190824", "YYYYMMDD").fromNow(),
                text: 'thanks',
                recieved: true
            }
        ]
    },
    {
        name: 'Kolya',
        online: 'last seen 10 minutes ago',
        id: 3,
        messages: [
            {
                id: 31,
                date: moment("20190826", "YYYYMMDD").fromNow(),
                text: 'YO!',
                recieved: false
            },
            {
                id: 32,
                date: moment("20190827", "YYYYMMDD").fromNow(),
                text: 'HI!',
                recieved: true
            }
        ]
    }
]