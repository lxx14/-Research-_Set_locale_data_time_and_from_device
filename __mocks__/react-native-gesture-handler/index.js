import React from 'react';
import { View } from 'react-native';

export const FlatList = () => {    //заглушка
    return <View />
}
export const RectButton = () => {
    return <View />
}