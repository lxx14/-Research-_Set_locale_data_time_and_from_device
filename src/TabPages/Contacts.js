import React, { Component } from 'react';

import { Text, View } from 'react-native';

import { styles } from './styles';

class Contacts extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <View style={styles.container}>
                <Text>Contacts</Text>
            </View>
        )
    }
}

export default Contacts;