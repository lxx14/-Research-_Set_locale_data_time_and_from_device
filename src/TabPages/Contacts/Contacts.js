import React, { Component } from 'react';
import { connect } from 'react-redux';
import { View } from 'react-native';

import { getUser } from '../../utils/getUser';
import { styles } from './styles';

export class Contacts extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { users } = this.props;

        return (
            <View style={styles.container}>
                {getUser(users, 'contacts', this.props)}
            </View>
        )
    }
}

const mapStateToProps = (state) => ({
    users: state.users
});

export default connect(mapStateToProps)(Contacts);