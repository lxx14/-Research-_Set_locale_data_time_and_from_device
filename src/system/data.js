export const users = [
    {
        name: 'Vasya',
        id: 1,
        messages: [
            {
                id: 11,
                text: 'hello',
                recieved: false
            },
            {
                id: 12,
                text: 'hi bro',
                recieved: true
            }
        ]
    },
    {
        name: 'Petya',
        id: 2,
        messages: [
            {
                id: 21,
                text: 'Have a nice day!',
                recieved: false
            },
            {
                id: 22,
                text: 'thanks',
                recieved: true
            }
        ]
    }
]