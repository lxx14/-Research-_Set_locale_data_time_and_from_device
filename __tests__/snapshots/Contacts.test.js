import React from 'react';
import { Contacts } from '../../src/TabPages/Contacts/Contacts';
import ShallowRenderer from 'react-test-renderer/shallow';
import { users } from '../../__mocks__/data';

//connected component to redux, необходимо в компоненте Chats экспортировать еще сам компонент и вверху теста подключать его через {}

describe('Contacts', () => {
    it('create snapshot Contacts', () => {
        const renderer = new ShallowRenderer()
        const snapshot = renderer.render(<Contacts users={users} />);
        expect(snapshot).toMatchSnapshot();
    });
})