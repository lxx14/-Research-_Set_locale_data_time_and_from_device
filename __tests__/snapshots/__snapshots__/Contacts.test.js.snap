// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts create snapshot Contacts 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Memo(Connect(User))
    chats={false}
    id={1}
    messages={null}
    name="Vasya"
    online="online"
    user={
      Object {
        "id": 1,
        "messages": Array [
          Object {
            "date": "11 days ago",
            "id": 11,
            "recieved": false,
            "text": "hello",
          },
          Object {
            "date": "9 days ago",
            "id": 12,
            "recieved": true,
            "text": "hi bro",
          },
        ],
        "name": "Vasya",
        "online": "online",
      }
    }
    users={
      Array [
        Object {
          "id": 1,
          "messages": Array [
            Object {
              "date": "11 days ago",
              "id": 11,
              "recieved": false,
              "text": "hello",
            },
            Object {
              "date": "9 days ago",
              "id": 12,
              "recieved": true,
              "text": "hi bro",
            },
          ],
          "name": "Vasya",
          "online": "online",
        },
        Object {
          "id": 2,
          "messages": Array [
            Object {
              "date": "10 days ago",
              "id": 21,
              "recieved": false,
              "text": "Have a nice day!",
            },
            Object {
              "date": "7 days ago",
              "id": 22,
              "recieved": true,
              "text": "thanks",
            },
          ],
          "name": "Petya",
          "online": "last seen just now",
        },
        Object {
          "id": 3,
          "messages": Array [
            Object {
              "date": "5 days ago",
              "id": 31,
              "recieved": false,
              "text": "YO!",
            },
            Object {
              "date": "4 days ago",
              "id": 32,
              "recieved": true,
              "text": "HI!",
            },
          ],
          "name": "Kolya",
          "online": "last seen 10 minutes ago",
        },
      ]
    }
  />
  <Memo(Connect(User))
    chats={false}
    id={2}
    messages={null}
    name="Petya"
    online="last seen just now"
    user={
      Object {
        "id": 2,
        "messages": Array [
          Object {
            "date": "10 days ago",
            "id": 21,
            "recieved": false,
            "text": "Have a nice day!",
          },
          Object {
            "date": "7 days ago",
            "id": 22,
            "recieved": true,
            "text": "thanks",
          },
        ],
        "name": "Petya",
        "online": "last seen just now",
      }
    }
    users={
      Array [
        Object {
          "id": 1,
          "messages": Array [
            Object {
              "date": "11 days ago",
              "id": 11,
              "recieved": false,
              "text": "hello",
            },
            Object {
              "date": "9 days ago",
              "id": 12,
              "recieved": true,
              "text": "hi bro",
            },
          ],
          "name": "Vasya",
          "online": "online",
        },
        Object {
          "id": 2,
          "messages": Array [
            Object {
              "date": "10 days ago",
              "id": 21,
              "recieved": false,
              "text": "Have a nice day!",
            },
            Object {
              "date": "7 days ago",
              "id": 22,
              "recieved": true,
              "text": "thanks",
            },
          ],
          "name": "Petya",
          "online": "last seen just now",
        },
        Object {
          "id": 3,
          "messages": Array [
            Object {
              "date": "5 days ago",
              "id": 31,
              "recieved": false,
              "text": "YO!",
            },
            Object {
              "date": "4 days ago",
              "id": 32,
              "recieved": true,
              "text": "HI!",
            },
          ],
          "name": "Kolya",
          "online": "last seen 10 minutes ago",
        },
      ]
    }
  />
  <Memo(Connect(User))
    chats={false}
    id={3}
    messages={null}
    name="Kolya"
    online="last seen 10 minutes ago"
    user={
      Object {
        "id": 3,
        "messages": Array [
          Object {
            "date": "5 days ago",
            "id": 31,
            "recieved": false,
            "text": "YO!",
          },
          Object {
            "date": "4 days ago",
            "id": 32,
            "recieved": true,
            "text": "HI!",
          },
        ],
        "name": "Kolya",
        "online": "last seen 10 minutes ago",
      }
    }
    users={
      Array [
        Object {
          "id": 1,
          "messages": Array [
            Object {
              "date": "11 days ago",
              "id": 11,
              "recieved": false,
              "text": "hello",
            },
            Object {
              "date": "9 days ago",
              "id": 12,
              "recieved": true,
              "text": "hi bro",
            },
          ],
          "name": "Vasya",
          "online": "online",
        },
        Object {
          "id": 2,
          "messages": Array [
            Object {
              "date": "10 days ago",
              "id": 21,
              "recieved": false,
              "text": "Have a nice day!",
            },
            Object {
              "date": "7 days ago",
              "id": 22,
              "recieved": true,
              "text": "thanks",
            },
          ],
          "name": "Petya",
          "online": "last seen just now",
        },
        Object {
          "id": 3,
          "messages": Array [
            Object {
              "date": "5 days ago",
              "id": 31,
              "recieved": false,
              "text": "YO!",
            },
            Object {
              "date": "4 days ago",
              "id": 32,
              "recieved": true,
              "text": "HI!",
            },
          ],
          "name": "Kolya",
          "online": "last seen 10 minutes ago",
        },
      ]
    }
  />
</View>
`;
